orderer/consensus/etcdraft/chain.go

```diff
@@ -1315,6 +1315,11 @@ func (c *Chain) ValidateConsensusMetadata(oldMetadataBytes, newMetadataBytes []b
 	return nil
 }
 
+// StatusReport returns the ClusterRelation & Status
+func (c *Chain) StatusReport() (string, string) {
+	return "member", "active"
```

Comment:
Perhaps instead of constructing the API results in a loose manner such as here, we define constants and return them?
That's also why I suggest using enums, so that you can have different types for "relation" and "status".  Also I am not sure why we need 2 different aspects. 
We have "member", and "follower" for a role, and a status.
Can't we flatten them with a Cartesian product of all possible combinations into a single enum? How many are there? I'm sure it's not too many...

Reply:
These go directly into the `types.ChannelInfo` struct which gets marshaled into json and sent in the http response. I do not see a point in having an intermediate type as an enum, and then translating to the strings that need to go into the json response.

As you can see in the documentation below, the "ClusterRelation" and "Status"  fields go into separate fields in that struct, which what the API design calls for. So the internal representation follows the API to avoid unnecessary translations. 

Note also the we have, for backwards compatibility, not just "member" & "follower" but also "config-tracker" & "none". The status is "active", "onboarding" and "inactive". So I wouldn't want to create all the combinations (as these might also evolve in the future).

```
type ChannelInfo struct {
	// The channel name.
	Name string `json:"name"`
	// The channel relative URL (no Host:Port, only path), e.g.: "/participation/v1/channels/my-channel".
	URL string `json:"url"`
	// Whether the orderer is a “member” or ”follower” of the cluster, or "config-tracker" of the cluster, for this channel.
	// For non cluster consensus types (solo, kafka) it is "none".
	// Possible values:  “member”, ”follower”, "config-tracker", "none".
	ClusterRelation string `json:"clusterRelation"`
	// Whether the orderer is ”onboarding”, ”active”, or "inactive", for this channel.
	// For non cluster consensus types (solo, kafka) it is "active".
	// Possible values:  “onboarding”, ”active”, "inactive".
	Status string `json:"status"`
	// Current block height.
	Height uint64 `json:"height"`
}
```

What I can do, and thought of doing but sustained, is to create constants in the types package, e.g.
```go
const (
	ChannelInfoClusterRelationMember string = "member"
)
```
and so on.

Reply:
Constants will do I guess :+1:

Reply:
or perhaps something like 

```go

type ClusterRelation string
const (
	ClusterRelationMember ClusterRelation = "member"
)
```

Reply:
Yes this is better